### PR TITLE
Precland Companion Computer support

### PR DESCRIPTION
**Pull request by djnugent**
-Precland now works with the LANDING_TARGET message.
-Updated LANDING_TARGET to match upstream mavlink
-Added a new PLAND_SCALE parameter to scale body angles from camera
-Added precland lidar support
-Fixed bug which acted on already consumed target data

### CODE REVIEW

**Comment by squilter** [comment]
Ayy!

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -432,7 +432,7 @@ void Copter::send_pid_tuning(mavlink_channel_t chan)
     const Vector3f &gyro = ahrs.get_gyro();
     if (g.gcs_pid_mask & 1) {
         const DataFlash_Class::PID_Info &pid_info = g.pid_rate_roll.get_pid_info();
-        mavlink_msg_pid_tuning_send(chan, PID_TUNING_ROLL, 
+        mavlink_msg_pid_tuning_send(chan, PID_TUNING_ROLL,
```
Comment (squilter) [inline ArduCopter/GCS_Mavlink.cpp]: Looks like your editor is messing with whitespace

**Comment by rmackay9** [comment]
merged, thanks!
I made a few changes as I merged it:
1. fixed the formatting (tabs vs spaces, removed unintended format changes)
2. split the commits up.  Normally we don't have a single commit affecting both the vehicle code and the libraries nor multiple libraries.
3. fixed up the use of the sonar altitude (removed the tilt correction because that should be done centrally if we're going to do that).
4. removed the SCALE parameter.  It's a good idea but won't be necessary once we move to the velocity controller.  Also if we're going to scale it, we need to scale it depending upon the update rate from the sensor.  Otherwise the scale value would need to be 5x higher for a sensor that updates 5x faster to achieve the same effect.

**Comment by rmackay9** [comment]
By the way, looking at how the companion computer is consuming the new samples, I can see that the IRLock equivalent has a problem in that it could very well consume the same sample more than once.  I'll tell the IRLock guys or maybe fix it myself some time.

**Comment by djnugent** [comment]
Sounds good. Thanks.

**Pull request closed without merging**